modules/extensions/src/main/java/org/apache/synapse/mediators/opa/OPAMediator.java

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.synapse.mediators.opa;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.mediators.AbstractMediator;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OPAMediator extends AbstractMediator {
+
+    private static final Log log = LogFactory.getLog(OPAMediator.class);
+
+    private String serverUrl = null;
+    private String accessToken = null;
+    private String policy = null;
+    private String rule = "allow";
+    private String requestGeneratorClassName = "org.apache.synapse.mediators.opa.OPASynapseRequestGenerator";
+    private Map<String, Object> advancedProperties = new HashMap<String, Object>();
+
+    public void init() {
+
+    }
+
+    @Override
+    public boolean mediate(MessageContext messageContext) {
+
+        OPARequestGenerator requestGenerator = null;
+        String opaResponseString = null;
+        try {
+            try {
+                Class<?> requestGeneratorClassObject = Class.forName(requestGeneratorClassName);
```

Comment:
don't we have the dependency for OPASynapseRequestGenerator here, why are we using dynamic loading?

Reply:
We provide the option to Implement the same interface from APIM side. In that scenario, we provide the class name as a mediator input and the request generator object is created from that implementation.